Added size comparison to xRemoteFile

Tiberriver256 opened the pull request:
When MatchSource is $False and Test-Path is $True added a webclient
content-length check to compare the size in bytes of the target and
destination files.

This will ensure that the file is re-downloaded if the file sizes are
not the same (i.e. target file has been updated, download was
interrupted previously, local file was corrupted, etc.)

msftclas commented:
Hi __@Tiberriver256__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

KarolKaczmarek commented on DSCResources/MSFT_xRemoteFile/MSFT_xRemoteFile.psm1:
```diff
@@ -253,7 +253,13 @@ function Test-TargetResource
                 }
             } else {
                 Write-Debug "MatchSource is false. No need for downloading file."
```
Please move this message before line:
```
$fileExists = $true
```
and add another one saying that file will be downloaded if size isn't the same.

msftclas commented:
Hi __@Tiberriver256__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

Tiberriver256 commented:
Sorry, I'm not sure what happened but I added in the Write-Debug lines and it somehow thinks I changed 400+ lines of code.

Should I close and re-open the pull request?

KarolKaczmarek commented:
if that isn't a problem, that would be great
